modules/common-fs2/src/main/scala/com/snowplowanalytics/snowplow/enrich/common/fs2/io/Clients.scala

```diff
@@ -69,6 +70,19 @@ object Clients {
       .withRequestTimeout(readTimeout)
       .withMaxTotalConnections(maxConnections)
       .resource
+      .map(Retry[F](retryPolicy, redactHeadersWhen))
+
+  private def retryPolicy[F[_]] =
+    RetryPolicy[F](
+      RetryPolicy.exponentialBackoff(maxWait = 10.seconds, maxRetry = 10),
+      retriable = {
+        case (_, Left(Command.EOF)) => true
+        case _ => false
+      }
+    )
+
+  private def redactHeadersWhen(header: CaseInsensitiveString) =
+    (Headers.SensitiveHeaders + "apikey".ci).contains(header)
```

Comment:
I assume this is because http4s automatically prints something to the logs in case of a retry.  Please can you share an example of what gets printed?  And which log level is it at?

Reply:
[Here](https://github.com/http4s/http4s/blob/382bcb8419cb050c7d38df1d17327da60e7e59b8/client/src/main/scala/org/http4s/client/middleware/Retry.scala#L40) is the code with logging. It's all info log level.

In the case of EOF and retry attempt:

```
[pool-1-thread-1] INFO org.http4s.client.middleware.Retry - Request threw an exception on attempt #1. Retrying after 0 milliseconds
EOF
```
Request is not printed. But looking at the code, when we exhaust attempts it would be printed with headers. It's very unlikely it happens, but still possible! 

When it actually happens, sensitive data is printed as: `apikey: <REDACTED>`